In [1]:
%pylab inline

Populating the interactive namespace from numpy and matplotlib


In [2]:
Ionpos_Methanediol_ang_nc=[[ 1.35170 ,  -0.08370,    0.00000  ], 
                    [-0.52950,    1.24650 ,  -0.00000  ], 
                    [-0.06950,   -0.09830,   -0.00000  ],
                    [-0.43450 ,  -0.61450  ,  0.88910  ],
                    [-0.43450 ,  -0.61450 ,  -0.88910 ],
                    [1.62470 ,  -1.02540  ,  0.00000  ],
                    [-1.50830 ,   1.19000  ,  0.00000]]
ions_Methanediol=['O','O','C','H','H','H','H']

In [3]:
Ionpos_Methanediol_ang_nr= Ionpos_Methanediol_ang_nc-np.array([-0.43450 ,  -0.61450 ,  -0.88910])  #shift H that doesnt belong to the origin
print(Ionpos_Methanediol_ang_nr)

[[ 1.7862  0.5308  0.8891]
 [-0.095   1.861   0.8891]
 [ 0.365   0.5162  0.8891]
 [ 0.      0.      1.7782]
 [ 0.      0.      0.    ]
 [ 2.0592 -0.4109  0.8891]
 [-1.0738  1.8045  0.8891]]


In [4]:
b=np.mat([0,0,1])
a=np.mat([ 0.365 ,  0.5162 , 0.8891]) #rotate to make z axis the one connecting the hydrogen to be removed and the carbon
a=a/np.sqrt(np.dot(a,a.T))
c=np.dot(a,b.T)
v=np.cross(a,b)[0]
s=np.sqrt(np.dot(v,v.T))
y = np.mat([[0, -v[2], v[1]], [v[2], 0, -v[0]], [-v[1], v[0], 0]])

Rot_Methanediol=np.mat(np.diag([1,1,1]))+y+np.dot(y,y)*(((1-c)/s**2)[0,0])
Ionpos_Methanediol_ang=np.array(np.dot(Rot_Methanediol,np.mat(Ionpos_Methanediol_ang_nr.T))).T

In [5]:
 Ionpos_Formaldehyde_ang_nc=np.array([[ 0.000000   ,  .000000   ,  .683501],
            [.000000   ,  .000000 ,   -.536614],
            [.000000  ,   .934390  , -1.124164],
            [.000000  ,  -.934390  , -1.124164]])
ions_Formaldehyde=['O','C','H','H']

In [6]:
Ionpos_Formaldehyde_ang_nr= Ionpos_Formaldehyde_ang_nc-np.array([.000000  ,  -.934390  , -1.124164]) #shift H that doesnt belong to the origin
print(Ionpos_Formaldehyde_ang_nr)

[[0.       0.93439  1.807665]
 [0.       0.93439  0.58755 ]
 [0.       1.86878  0.      ]
 [0.       0.       0.      ]]


In [7]:
b=np.mat([0,0,1])
a=np.mat([0.     ,  0.93439 , 0.58755 ]) #rotate to make z axis the one connecting the hydrogen to be removed and the carbon
a=a/np.sqrt(np.dot(a,a.T))
c=np.dot(a,b.T)
v=np.cross(a,b)[0]
s=np.sqrt(np.dot(v,v.T))
y = np.mat([[0, -v[2], v[1]], [v[2], 0, -v[0]], [-v[1], v[0], 0]])

Rot_Formaldehyde=np.mat(np.diag([1,1,1]))+y+np.dot(y,y)*(((1-c)/s**2)[0,0])
Ionpos_Formaldehyde_ang=np.array(np.dot(Rot_Formaldehyde,np.mat(Ionpos_Formaldehyde_ang_nr.T))).T


In [8]:
#attempt at creating the Pt file
Coords1=(np.linspace(0,0.5,2))/2
Coords2=(np.linspace(0,0.5,2))/2
Coords3=(np.linspace(-0.2,.2,5))
print(Coords1,Coords2,Coords3)

[0.   0.25] [0.   0.25] [-0.2 -0.1  0.   0.1  0.2]


In [9]:
Ionpos_Pt=[[i,j,k] for i in Coords1 for j in Coords2 for k in Coords3]
ions_Pt=['Pt' for i in range(len(Ionpos_Pt))]

In [10]:
for i in range(len(Ionpos_Pt)):
    print('Ion '+ions_Pt[i], end=' ' )
    for j in range(len(Ionpos_Pt[i])):
        print("%.3f" % Ionpos_Pt[i][j], end = ' ')
    print('\n')

Ion Pt 0.000 0.000 -0.200 

Ion Pt 0.000 0.000 -0.100 

Ion Pt 0.000 0.000 0.000 

Ion Pt 0.000 0.000 0.100 

Ion Pt 0.000 0.000 0.200 

Ion Pt 0.000 0.250 -0.200 

Ion Pt 0.000 0.250 -0.100 

Ion Pt 0.000 0.250 0.000 

Ion Pt 0.000 0.250 0.100 

Ion Pt 0.000 0.250 0.200 

Ion Pt 0.250 0.000 -0.200 

Ion Pt 0.250 0.000 -0.100 

Ion Pt 0.250 0.000 0.000 

Ion Pt 0.250 0.000 0.100 

Ion Pt 0.250 0.000 0.200 

Ion Pt 0.250 0.250 -0.200 

Ion Pt 0.250 0.250 -0.100 

Ion Pt 0.250 0.250 0.000 

Ion Pt 0.250 0.250 0.100 

Ion Pt 0.250 0.250 0.200 



In [11]:
Coords1=(np.linspace(0,0.5,2))
Coords2=(np.linspace(0,0.5,2))
Coords3=(np.linspace(-0.2,.2,5))
print(Coords1,Coords2,Coords3)

[0.  0.5] [0.  0.5] [-0.2 -0.1  0.   0.1  0.2]


In [12]:
Ionpos_Pt=[[i,j,k] for i in Coords1 for j in Coords2 for k in Coords3]
ions_Pt=['Pt' for i in range(len(Ionpos_Pt))]

In [13]:
for i in range(len(Ionpos_Pt)):
    print('Ion '+ions_Pt[i], end=' ' )
    for j in range(len(Ionpos_Pt[i])):
        print("%.3f" % Ionpos_Pt[i][j], end = ' ')
    print('\n')

Ion Pt 0.000 0.000 -0.200 

Ion Pt 0.000 0.000 -0.100 

Ion Pt 0.000 0.000 0.000 

Ion Pt 0.000 0.000 0.100 

Ion Pt 0.000 0.000 0.200 

Ion Pt 0.000 0.500 -0.200 

Ion Pt 0.000 0.500 -0.100 

Ion Pt 0.000 0.500 0.000 

Ion Pt 0.000 0.500 0.100 

Ion Pt 0.000 0.500 0.200 

Ion Pt 0.500 0.000 -0.200 

Ion Pt 0.500 0.000 -0.100 

Ion Pt 0.500 0.000 0.000 

Ion Pt 0.500 0.000 0.100 

Ion Pt 0.500 0.000 0.200 

Ion Pt 0.500 0.500 -0.200 

Ion Pt 0.500 0.500 -0.100 

Ion Pt 0.500 0.500 0.000 

Ion Pt 0.500 0.500 0.100 

Ion Pt 0.500 0.500 0.200 



In [14]:
for i in range(len(Ionpos_Formaldehyde_ang)):
    print('Ion '+ions_Formaldehyde[i], end=' ' )
    for j in range(len(Ionpos_Formaldehyde_ang[i])):
        print("%.3f" % Ionpos_Formaldehyde_ang[i][j], end = ' ')
    print('\n')

Ion O 0.000 -1.033 1.753 

Ion C 0.000 0.000 1.104 

Ion H 0.000 0.995 1.582 

Ion H 0.000 0.000 0.000 



In [15]:
for i in range(len(Ionpos_Methanediol_ang)):
    print('Ion '+ions_Methanediol[i], end=' ' )
    for j in range(len(Ionpos_Methanediol_ang[i])):
        print("%.3f" % Ionpos_Methanediol_ang[i][j], end = ' ')
    print('\n')

Ion O 1.332 -0.111 1.573 

Ion O -0.549 1.219 1.573 

Ion C 0.000 0.000 1.091 

Ion H -0.595 -0.841 1.449 

Ion H 0.000 0.000 0.000 

Ion H 1.671 -0.961 1.219 

Ion H -1.462 1.255 1.219 



In [16]:
Ionpos_Formaldehyde_p=np.array(Ionpos_Formaldehyde_ang)-array(Ionpos_Formaldehyde_ang[-1])
Ionpos_Methanediol_p=np.array(Ionpos_Methanediol_ang)-array(Ionpos_Methanediol_ang[4])
for i in range(len(Ionpos_Formaldehyde_p)):
    print('Ion '+ions_Formaldehyde[i], end=' ' )
    for j in range(len(Ionpos_Formaldehyde_p[i])):
        print("%.3f" % Ionpos_Formaldehyde_p[i][j], end = ' ')
    print('\n')
print('__________________________')
for i in range(len(Ionpos_Methanediol_p)):
    print('Ion '+ions_Methanediol[i], end=' ' )
    for j in range(len(Ionpos_Methanediol_p[i])):
        print("%.3f" % Ionpos_Methanediol_p[i][j], end = ' ')
    print('\n')

Ion O 0.000 -1.033 1.753 

Ion C 0.000 0.000 1.104 

Ion H 0.000 0.995 1.582 

Ion H 0.000 0.000 0.000 

__________________________
Ion O 1.332 -0.111 1.573 

Ion O -0.549 1.219 1.573 

Ion C 0.000 0.000 1.091 

Ion H -0.595 -0.841 1.449 

Ion H 0.000 0.000 0.000 

Ion H 1.671 -0.961 1.219 

Ion H -1.462 1.255 1.219 



In [17]:
Ionpos_Formaldehyde_p=Ionpos_Formaldehyde_p #2.05 is the bond length of carbon and Pt
Ionpos_Methanediol_p=Ionpos_Methanediol_p

In [18]:
#conversion factors for orthorhombic lattice
a_Pt=0.392 #nm
a_Pt_ang=a_Pt*10 #angstrom
a_Pt_ang1=2*a_Pt_ang/sqrt(2) #angstrom
a_Pt_ang2=2*a_Pt_ang/sqrt(2/3) #angstrom
a_Pt_ang3=5*(sqrt(3)*a_Pt_ang/3)+5*(sqrt(3)*a_Pt_ang/3)  #angstrom (each layer is sqrt(3)a/3 ) 5 layers of material and 5 of separation


#conversion factors for orthorhombic lattice
a_Pt=0.392 #nm
a_Pt_ang=a_Pt*10 #angstrom
a_Pt_ang1=2*a_Pt_ang*sqrt(2) #angstrom
a_Pt_ang2=2*a_Pt_ang*sqrt(2) #angstrom
a_Pt_ang3=5*(sqrt(3)*a_Pt_ang/3)+5*(sqrt(3)*a_Pt_ang/3)  #angstrom (each layer is sqrt(3)a/3 ) 5 layers of material and 5 of separation


print(a_Pt_ang3,a_Pt_ang1*1.88973,a_Pt_ang2*1.88973,a_Pt_ang3*1.88973)
#A100_Formaldehyde=np.array([np.ones(len(Ionpos_Formaldehyde_p))*a_Pt_ang1,np.ones(len(Ionpos_Formaldehyde_p))*a_Pt_ang2,np.ones(len(Ionpos_Formaldehyde_p))*a_Pt_ang3])
#A100_Ionpos_Methanediol=np.array([np.ones(len(Ionpos_Methanediol_p))*a_Pt_ang1,np.ones(len(Ionpos_Methanediol_p))*a_Pt_ang2,np.ones(len(Ionpos_Methanediol_p))*a_Pt_ang3])



### change of basis from rotated hexagonal to euclidean (this is just stacking the basis vectors of the hexagonal 
#### side by side)
A100_CHangebasis_inv=(mat(array([[0 , -a_Pt_ang/2 , a_Pt_ang],[a_Pt_ang/2  , 0 ,  a_Pt_ang],[-a_Pt_ang/2 , a_Pt_ang/2 , a_Pt_ang]])))

###rotating the vectors found in the tutorial (rotated hexagonal that points in the 111 direction) 
### to get the hexagonal lattice vectors perpendicular to the z axis to setup the molecule coordinates

b=np.mat([0,0,1])
a=np.mat([1 ,  1 , 1 ]) #rotate to make z axis the 111 direction
a=a/np.sqrt(np.dot(a,a.T))
c=np.dot(a,b.T)
v=np.cross(a,b)[0]
s=np.sqrt(np.dot(v,v.T))
y = np.mat([[0, -v[2], v[1]], [v[2], 0, -v[0]], [-v[1], v[0], 0]])

Rot=np.mat(np.diag([1,1,1]))+y+np.dot(y,y)*(((1-c)/s**2)[0,0]) ## rotates to hexagonal with z axis as the c axis
A100_CHangebasis_inv_True=np.array(np.dot(Rot,A100_CHangebasis_inv)) ### change of basis from hexagonal to euclidean

##scaling the third coordinate
A100_CHangebasis_inv_scaled=dot(diag([1,1,a_Pt_ang3/A100_CHangebasis_inv_True[2][2]]),A100_CHangebasis_inv_True)

A100_CHangebasis=inv(A100_CHangebasis_inv_True) ##change of basis from  euclidean to hexagonal


22.632130552233328 20.95225727455074 20.95225727455074 42.768616068471886


In [19]:
Ionpos_Formaldehyde_pp=array((dot(A100_CHangebasis,Ionpos_Formaldehyde_p.T)).T)

In [20]:
Ionpos_Methanediol_pp=array((dot(A100_CHangebasis,Ionpos_Methanediol_p.T)).T)

In [21]:
for i in range(len(Ionpos_Formaldehyde_pp)):
    print('ion '+ions_Formaldehyde[i], end=' ' )
    for j in range(len(Ionpos_Formaldehyde_pp[i])):
        print("%.3f" % Ionpos_Formaldehyde_pp[i][j], end = ' ')
    print('1 \n')
print('__________________________')
for i in range(len(Ionpos_Methanediol_pp)):
    print('ion '+ions_Methanediol[i], end=' ' )
    for j in range(len(Ionpos_Methanediol_pp[i])):
        print("%.3f" % Ionpos_Methanediol_pp[i][j], end = ' ')
    print('1 \n')

ion O -0.416 -0.111 0.258 1 

ion C 0.000 -0.000 0.163 1 

ion H 0.400 0.107 0.233 1 

ion H 0.000 0.000 0.000 1 

__________________________
ion O -0.188 -0.548 0.232 1 

ion O 0.550 0.352 0.232 1 

ion C 0.000 -0.000 0.161 1 

ion H -0.274 0.149 0.213 1 

ion H 0.000 0.000 0.000 1 

ion H -0.567 -0.776 0.180 1 

ion H 0.663 0.724 0.180 1 



In [22]:
floor((5*(sqrt(3)*a_Pt_ang/3))*1.88973 +15)

36.0

In [23]:
#distance to platinum
Dis_ang=.186*10 #literature value in angstrom
Dis=Dis_ang/a_Pt_ang3 #in fractional coordinates 


#summing the position of the central atom in the top most layer of  the surface

Dis

0.08218404342036

In [24]:
Ionpos_Formaldehyde_ppp=Ionpos_Formaldehyde_pp+np.array([0,0,Dis-0.163])+np.array([0.5,0.5, 0]) #substracting top carbon pos after rescaling to 6 separation layers then adding top PT and bond distance from literature
Ionpos_Methanediol_ppp=Ionpos_Methanediol_pp+np.array([0,0,Dis-0.163  ])+np.array([0.5,0.5, 0]) #substracting top carbon pos then adding top PT and bond distance from literature

In [25]:
for i in range(len(Ionpos_Formaldehyde_ppp)):
    print('ion '+ions_Formaldehyde[i], end=' ' )
    for j in range(len(Ionpos_Formaldehyde_ppp[i])):
        print("%.3f" % Ionpos_Formaldehyde_ppp[i][j], end = ' ')
    print(' 1\n')
print('__________________________')
for i in range(len(Ionpos_Methanediol_ppp)):
    print('ion '+ions_Methanediol[i], end=' ' )
    for j in range(len(Ionpos_Methanediol_ppp[i])):
        print("%.3f" % Ionpos_Methanediol_ppp[i][j], end = ' ')
    print(' 1\n')

ion O 0.084 0.389 0.177  1

ion C 0.500 0.500 0.082  1

ion H 0.900 0.607 0.152  1

ion H 0.500 0.500 -0.081  1

__________________________
ion O 0.312 -0.048 0.151  1

ion O 1.050 0.852 0.151  1

ion C 0.500 0.500 0.080  1

ion H 0.226 0.649 0.133  1

ion H 0.500 0.500 -0.081  1

ion H -0.067 -0.276 0.099  1

ion H 1.163 1.224 0.099  1



ion Pt 0.000 0.000 -0.200 1
ion Pt 0.500 0.500 -0.200 1
ion Pt 0.000 0.50 -0.200 1
ion Pt 0.50 0.000 -0.200 1
ion Pt 0.750 0.250 -0.100 1 
ion Pt 0.250 0.750 -0.100 1
ion Pt 0.250 0.250 -0.100 1
ion Pt 0.750 0.750 -0.100 1
ion Pt 0.50 0.000 0.000 1
ion Pt 0.000 0.000 0.000 1 
ion Pt 0.000 0.50 0.000 1 
ion Pt 0.500 0.50 0.000 1 
ion Pt 0.750 0.250 0.100 1 
ion Pt 0.250 0.750 0.100 1
ion Pt 0.250 0.250 0.100 1
ion Pt 0.750 0.750 0.100 1
ion Pt 0.000 0.000 0.200 1
ion Pt 0.000 0.50 0.200 1
ion Pt 0.50 0.000 0.200 1
ion Pt 0.50 0.50 0.200 1

In [26]:
np.array([-0.2,-0.1,0,.1,.2])*5/7

array([-0.14285714, -0.07142857,  0.        ,  0.07142857,  0.14285714])

ion Pt 0.000 0.000 -0.1428 1
ion Pt 0.500 0.500 -0.1428 1
ion Pt 0.000 0.50 -0.1428 1
ion Pt 0.50 0.000 -0.1428 1
ion Pt 0.750 0.250 -0.0714 1 
ion Pt 0.250 0.750 -0.0714 1
ion Pt 0.250 0.250 -0.0714 1
ion Pt 0.750 0.750 -0.0714 1
ion Pt 0.50 0.000 0.000 1
ion Pt 0.000 0.000 0.000 1 
ion Pt 0.000 0.50 0.000 1 
ion Pt 0.500 0.50 0.000 1 
ion Pt 0.750 0.250 0.0714 1 
ion Pt 0.250 0.750 0.0714 1
ion Pt 0.250 0.250 0.0714 1
ion Pt 0.750 0.750 0.0714 1
ion Pt 0.000 0.000 0.1428 1
ion Pt 0.000 0.50 0.1428 1
ion Pt 0.50 0.000 0.1428 1
ion Pt 0.50 0.50 0.1428 1

In [27]:
#for six layer unit cell
np.array([-0.187130,-0.093565,0.000000,0.093565,0.187130])*5/6

array([-0.15594167, -0.07797083,  0.        ,  0.07797083,  0.15594167])

ion Pt 0.000 0.000 -0.1667 1
ion Pt 0.500 0.500 -0.1667 1
ion Pt 0.000 0.50 -0.1667 1
ion Pt 0.50 0.000 -0.1667 1
ion Pt 0.750 0.250 -0.0833 1 
ion Pt 0.250 0.750 -0.0833 1
ion Pt 0.250 0.250 -0.0833 1
ion Pt 0.750 0.750 -0.0833 1
ion Pt 0.50 0.000 0.000 1
ion Pt 0.000 0.000 0.000 1 
ion Pt 0.000 0.50 0.000 1 
ion Pt 0.500 0.50 0.000 1 
ion Pt 0.750 0.250 0.0833 1 
ion Pt 0.250 0.750 0.0833 1
ion Pt 0.250 0.250 0.0833 1
ion Pt 0.750 0.750 0.0833 1
ion Pt 0.000 0.000 0.1428 1
ion Pt 0.000 0.50 0.1667 1
ion Pt 0.50 0.000 0.1667 1
ion Pt 0.50 0.50 0.1667 1

In [28]:
#for the carbon monoxide attached to the surface
OCdist=1.12/a_Pt_ang3
print(OCdist+0.268 )


0.3174871659305394


In [29]:
OCdist

0.04948716593053936

In [30]:
a_Pt_ang3


22.632130552233328

In [49]:
Ptpos=loadtxt('fracpos.dat')
print(Ptpos/2, size(Ptpos)/3)

[[ 0.3333335  0.1666665  0.262324 ]
 [ 0.3333335  0.6666665  0.262324 ]
 [ 0.8333335  0.1666665  0.262324 ]
 [ 0.8333335  0.6666665  0.262324 ]
 [-0.        -0.         0.321743 ]
 [ 0.         0.5        0.321743 ]
 [ 0.         1.         0.321743 ]
 [ 0.5       -0.         0.321743 ]
 [ 0.5        0.5        0.321743 ]
 [ 0.5        1.         0.321743 ]
 [ 1.        -0.         0.321743 ]
 [ 1.         0.5        0.321743 ]
 [ 1.         1.         0.321743 ]
 [ 0.1666665  0.3333335  0.381162 ]
 [ 0.1666665  0.8333335  0.381162 ]
 [ 0.6666665  0.3333335  0.381162 ]
 [ 0.6666665  0.8333335  0.381162 ]
 [ 0.3333335  0.1666665  0.440581 ]
 [ 0.3333335  0.6666665  0.440581 ]
 [ 0.8333335  0.1666665  0.440581 ]
 [ 0.8333335  0.6666665  0.440581 ]
 [ 0.         0.         0.       ]
 [-0.        -0.         0.5      ]
 [ 0.         0.5        0.       ]
 [ 0.         0.5        0.5      ]
 [ 0.         1.         0.       ]
 [ 0.         1.         0.5      ]
 [ 0.5        0.         0. 

In [50]:
Ptpos=(dot(bmat(diag([1/2,1/2,1])),Ptpos.T)).T

In [51]:
postPt=[]
for i in Ptpos:
    
    if 1. in i:
        print(i)
    else:
        postPt.append(array(i)[0])

[[0.       1.       0.643486]]
[[0.5      1.       0.643486]]
[[1.       0.       0.643486]]
[[1.       0.5      0.643486]]
[[1.       1.       0.643486]]
[[0. 0. 1.]]
[[0.  0.5 1. ]]
[[0. 1. 0.]]
[[0. 1. 1.]]
[[0.5 0.  1. ]]
[[0.5 0.5 1. ]]
[[0.5 1.  0. ]]
[[0.5 1.  1. ]]
[[1. 0. 0.]]
[[1. 0. 1.]]
[[1.  0.5 0. ]]
[[1.  0.5 1. ]]
[[1. 1. 0.]]
[[1. 1. 1.]]


In [52]:
for i in range(len(postPt)):
    print('ion Pt', end=' ' )
    for j in range(size(postPt[i])):
        print("%.4f" % postPt[i][j], end = ' ')
    print('1 \n')

ion Pt 0.3333 0.1667 0.5246 1 

ion Pt 0.3333 0.6667 0.5246 1 

ion Pt 0.8333 0.1667 0.5246 1 

ion Pt 0.8333 0.6667 0.5246 1 

ion Pt 0.0000 0.0000 0.6435 1 

ion Pt 0.0000 0.5000 0.6435 1 

ion Pt 0.5000 0.0000 0.6435 1 

ion Pt 0.5000 0.5000 0.6435 1 

ion Pt 0.1667 0.3333 0.7623 1 

ion Pt 0.1667 0.8333 0.7623 1 

ion Pt 0.6667 0.3333 0.7623 1 

ion Pt 0.6667 0.8333 0.7623 1 

ion Pt 0.3333 0.1667 0.8812 1 

ion Pt 0.3333 0.6667 0.8812 1 

ion Pt 0.8333 0.1667 0.8812 1 

ion Pt 0.8333 0.6667 0.8812 1 

ion Pt 0.0000 0.0000 0.0000 1 

ion Pt 0.0000 0.5000 0.0000 1 

ion Pt 0.5000 0.0000 0.0000 1 

ion Pt 0.5000 0.5000 0.0000 1 



In case shit hits the fan

ion Pt -0.333333  0.333333 -0.475352   1
ion Pt  0.000000  0.000000 -0.356514   1
ion Pt  0.333333 -0.333333 -0.237676   1
ion Pt -0.333333  0.333333 -0.118838   1
ion Pt  0.000000  0.000000  0.000000   1

In [53]:
import statistics
postPt=array(postPt)
postPt[-4:,2]=postPt[-4:,2]+1
centval=statistics.median(array(postPt)[:,2])
postPt[:,2]=postPt[:,2]-centval

Ptcoords=(array(dot(diag([1,1,floor((5*(sqrt(3)*a_Pt_ang/3))*1.88973 +15)/(a_Pt_ang3*1.88973)]),mat(postPt).T).T)) 
#Ptcoords=list(array(dot(diag([1,1,1]),mat(postPt).T).T)) 
centval=statistics.median((Ptcoords)[:,2])
Ptcoords[:,2]=Ptcoords[:,2]-max((Ptcoords)[:,2])
## the weird expression 
#in the denominator is shankars choice 
#for cleaving the surface, leaving a 15 bohr separation between slabs and 5 layers of thickness sqrt3*a/3 each
## I work first in angstrom and that is why I put the 1.88973


In [54]:
Ptcoords

array([[ 0.3333335 ,  0.1666665 , -0.40012218],
       [ 0.3333335 ,  0.6666665 , -0.40012218],
       [ 0.8333335 ,  0.1666665 , -0.40012218],
       [ 0.8333335 ,  0.6666665 , -0.40012218],
       [ 0.        ,  0.        , -0.30009164],
       [ 0.        ,  0.5       , -0.30009164],
       [ 0.5       ,  0.        , -0.30009164],
       [ 0.5       ,  0.5       , -0.30009164],
       [ 0.1666665 ,  0.3333335 , -0.20006109],
       [ 0.1666665 ,  0.8333335 , -0.20006109],
       [ 0.6666665 ,  0.3333335 , -0.20006109],
       [ 0.6666665 ,  0.8333335 , -0.20006109],
       [ 0.3333335 ,  0.1666665 , -0.10003055],
       [ 0.3333335 ,  0.6666665 , -0.10003055],
       [ 0.8333335 ,  0.1666665 , -0.10003055],
       [ 0.8333335 ,  0.6666665 , -0.10003055],
       [ 0.        ,  0.        ,  0.        ],
       [ 0.        ,  0.5       ,  0.        ],
       [ 0.5       ,  0.        ,  0.        ],
       [ 0.5       ,  0.5       ,  0.        ]])

In [55]:
for i in range(len(Ptcoords)):
    print('ion Pt', end=' ' )
    for j in range(size(Ptcoords[i])):
        print("%.4f" % Ptcoords[i][j], end = ' ')
    print('1 ')

ion Pt 0.3333 0.1667 -0.4001 1 
ion Pt 0.3333 0.6667 -0.4001 1 
ion Pt 0.8333 0.1667 -0.4001 1 
ion Pt 0.8333 0.6667 -0.4001 1 
ion Pt 0.0000 0.0000 -0.3001 1 
ion Pt 0.0000 0.5000 -0.3001 1 
ion Pt 0.5000 0.0000 -0.3001 1 
ion Pt 0.5000 0.5000 -0.3001 1 
ion Pt 0.1667 0.3333 -0.2001 1 
ion Pt 0.1667 0.8333 -0.2001 1 
ion Pt 0.6667 0.3333 -0.2001 1 
ion Pt 0.6667 0.8333 -0.2001 1 
ion Pt 0.3333 0.1667 -0.1000 1 
ion Pt 0.3333 0.6667 -0.1000 1 
ion Pt 0.8333 0.1667 -0.1000 1 
ion Pt 0.8333 0.6667 -0.1000 1 
ion Pt 0.0000 0.0000 0.0000 1 
ion Pt 0.0000 0.5000 0.0000 1 
ion Pt 0.5000 0.0000 0.0000 1 
ion Pt 0.5000 0.5000 0.0000 1 


In [38]:
centval

0.762324